lib/src/model/package_builder.dart

```diff
@@ -156,17 +154,21 @@ class PubPackageBuilder implements PackageBuilder {
 
     var analysisContext = contextCollection.contextFor(config.inputDir);
     var session = analysisContext.currentSession;
-    var sourceKind = await session.getSourceKind(filePath);
+    var isPart = false;
+    var sourceFile = session.getFile2(filePath);
+    if (sourceFile is FileResult) {
+      isPart = sourceFile.isPart;
```

Comment:
You don't need this.
`getResolvedLibrary2()` will return `NotLibraryButPartResult` if the path is a file that is a part.